### PR TITLE
Changes how ranks are assigned

### PR DESCRIPTION
**Pull request by Vasily2013**
## About The Pull Request

Stops renaming everyone to display their rank (_job.dm) and instead does it in speech (say.dm) and examine (examine.dm)

## Why It's Good For The Game

Cleaner way of doing it, can stop driving Francium up the wall

## Changelog
:cl:
refactor: Moved rank display out of _job.dm and into say.dm and examine.dm
/:cl:

### CODE REVIEW

**Comment by Kmc2000** [comment]
based change

**Comment by Shadow-Quill** [comment]
love it

**Comment by francinum** [comment]
@ConvictConclave-Bot fuck off ya bitch

File: code/game/say.dm [hunk 1]
```diff
@@ -71,14 +71,35 @@ GLOBAL_LIST_INIT(freqtospan, list(
...
+
+	if (istype(speaker, /mob/living/carbon/human))
+		var/mob/living/carbon/human/speakerMob = speaker
+		job = speakerMob.get_assignment()
+	else if (istype(speaker, /atom/movable/virtualspeaker))
+		var/atom/movable/virtualspeaker/VS = speaker
+		job = VS.GetJob()
+
+	if (job)
+		rank = "[SSjob.GetJob(job).display_rank] "
```
Comment (Kmc2000) [inline code/game/say.dm]:
```suggestion
		rank = "[SSjob.GetJob(job)?.display_rank] "
```

File: code/game/say.dm [hunk 2]
```diff
@@ -71,14 +71,35 @@ GLOBAL_LIST_INIT(freqtospan, list(
 	if(istype(D) && D.display_icon(src))
 		languageicon = "[D.get_icon()] "
 
-	return "[spanpart1][spanpart2][freqpart][languageicon][compose_track_href(speaker, namepart)][namepart][compose_job(speaker, message_language, raw_message, radio_freq)][endspanpart][messagepart]"
+	//NSV13 - added [compose_rank(speaker)]
+	return "[spanpart1][spanpart2][freqpart][languageicon][compose_track_href(speaker, namepart)][compose_rank(speaker)][namepart][compose_job(speaker, message_language, raw_message, radio_freq)][endspanpart][messagepart]"
```
Comment (Kmc2000) [inline code/game/say.dm]: Doesn't compose_rank return GSGT Sergei Koralev, so then compose rank + namepart returns GSGT Sergei Koralev Sergei Koralev?

Unless i'm misreading this code

Comment (covertcorvid) [reply code/game/say.dm]: Nah, compose_rank doesn't involve the name

**Pull request merged**